build: openpty is declared in different headers on some systems

Check for util.h next to pty.h for openpty is declared in it on at least
Darwin systems.  Solaris needs termios.h

--- telnet-bsd-1.2/configure.in	2024-07-26 10:13:46.814937718 +0200
+++ telnet-bsd-1.2/configure.in	2024-07-26 10:19:15.734016104 +0200
@@ -12,7 +12,7 @@
 AC_SUBST(PACKAGE)
 AC_SUBST(VERSION)
 
-AC_CHECK_HEADERS([pty.h utmp.h])
+AC_CHECK_HEADERS([pty.h util.h termios.h utmp.h])
 
 if test x"$ac_default_prefix" = x"/usr"
 then
--- telnet-bsd-1.2/telnetd/sys_term.c	2024-07-26 10:13:46.859137256 +0200
+++ telnet-bsd-1.2/telnetd/sys_term.c	2024-07-26 10:19:54.700983107 +0200
@@ -36,6 +36,12 @@
 #ifdef HAVE_PTY_H
 #include <pty.h>
 #endif
+#ifdef HAVE_UTIL_H
+#include <util.h>
+#endif
+#ifdef HAVE_TERMIOS_H
+#include <termios.h>
+#endif
 
 #include "telnetd.h"
 #include "pathnames.h"
